Lecture 4: Exact inference (#4)
sailinglab/pgm-spring-2019: _posts/2019-01-28-lecture-04.md

date: 2019-01-28

lecturers:
  - name: Eric Xing
    url: "https://www.cs.cmu.edu/~epxing/"

authors:
  - name: Tingfung Lau  # author's full name; let's order by sections
    url: "#"  # optional URL to the author's homepage
  - name: Austin Dill
    url: "#"
  - name: Lingxiao Zhao
    url: "#"
  - name: Ksenia Korovina
    url: "#"

editors:
  - name: Xun Zheng  # editor's full name
    url: "#"  # optional URL to the editor's homepage

abstract: >
  An example abstract block.
---

## Introduction
In this previous lectures, we introduce the concept of **Graphical Models** and its mathematical formulations. 
Now we know that we can use a graphical model $M$ (Bayesian network or undirected graph model) to specify a probability distribution $P_{M}$ satisfying some conditional independence property. 
In this lecture, we will study how to utilize a graphical model. Given a GM $M$, we generally have two type of tasks 
+ **Inference:** answering queries about the probability distribution $P_M$ defined by $M$, for examples, <script type="math/tex">P_M(X|Y)</script> where $X$ and $Y$ are subsets of variables in GM $M$.

> + **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning is task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 

Reviewer: ```suggestion
+ **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 
```

> + **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning is task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 

Reviewer: ```suggestion
+ **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 
```

The information flow between variables is not restricted by the directionality of the edges in a GM. 
We can actually do a probabilistic inference combing evidence from all parts of the networks.
Deep Belief Network (DBN) \[Hinton, 2006\]<d-cite key="hinton2006reducing"></d-cite> is an example. 
DBN is a generative model or Restricted Boltzmann Machine (RBM) with multiple layers. 
The model is successful for solving tasks like recognizing handwritten digits, learning motion capture data, collaborative filtering. 
The following figures shows a DBN with 3 hidden layers. We can infer hidden unit $H_1, H_2, H_3$ from data $V$. We can also generate data $V$ by sampling hidden units $H_3, H_2, H_1$ in the opposite direction.
<figure>
  <div class="row">
    <div class="col one">
      <img src="{{ '/assets/img/notes/lecture-04/dbn.png' | relative_url }}" />
    </div>
  </div>
  <figcaption>
    <strong>A Deep Belief Network with 3 hidden layers for image processing. </strong> 
  </figcaption>
</figure>

#### Most Probable Assignment
Another interesting query is to find the **most probable joint assignment**(MPA) for *some* variables in interest. 

> Such reasoning is usually performed under some evidence $\mathbf{e}$ and ignoring some "don'care" variables $Z$,

Reviewer: ```suggestion
Such reasoning is usually performed under some evidence $\mathbf{e}$ and ignoring some "don't care" variables $Z$,
```

The proof of the theorem can be found in Chapter 9 of Koller’s PGM textbook. This theorem tells us the scope of intermediate factors which is a elimination clique, is a clique in reconstituted graph. What’s more, the scope of the largest intermediate factor is a the largest maximal clique in reconstituted graph.

### Complexity Analysis in Graph Perspective

In the beginning of this section we have argued that the bottleneck of VE’s complexity is determined by the scope size of the maximum intermediate factor generated in the procedure of VE. In above subsection we have shown that each intermediate factor in VE is an elimination clique in graph elimination algorithm, and the largest elimination clique is also a largest maximal clique in reconstituted graph.

Then, given an elimination ordering $I$, the problem of get the bottleneck maximum scope size $N_{max}$ of intermediate factors is equivalent of finding the largest clique in reconstituted graph $G’_I(V, E’)$, which is a pure graph theoretic question. And there are efficient mature algorithm for solving this problem. 

### Elimination Ordering 

> We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G’_I(V, E’)$. For different ordering $I$, $w_{G,I}$ is different. 

Reviewer: ```suggestion
We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G_I'(V, E’)$. For different ordering $I$, $w_{G,I}$ is different. 
```

Reviewer: Not sure why is this breaking up in the website..

Author: We couldn't figure that out either, hopefully your suggestion helps.
